modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/store/ConnectedComponent.es.js

```diff
@@ -88,5 +90,48 @@ const getConnectedComponent = (Component, properties) => {
 	return ConnectedComponent;
 };
 
-export {getConnectedComponent};
+/**
+ * Second order function to produce a Connected Component Wrapper
+ * @param {Function} mapStateToProps - Recieves the state and returns mapped version of it ready for consumption by the Wrapped Component
+ * @param {Function} mapDispatchToProps - Recieves the dispatch and returns a set of component props that use it to call the modify the state tree
+ * @returns
+ */
+function getConnectedReactComponent(mapStateToProps, mapDispatchToProps) {
+	/**
+	 *
+	 * @param {React.Component} WrappedComponent - The Component to connect to the store
+	 * @returns {React.Component} - Wrapper Connected Component that propagates every store change mapped to the component
+	 */
+	return function _getConnectedWrapperComponent(WrappedComponent) {
+		return props => {
+			const store = useContext(StoreContext);
+
+			const [storeState, setStoreState] = useState(
+				store ? store.getState() : {}
+			);
+
+			useEffect(() => {
+				if (store) {
+					const handleStoreChange = () =>
+						setStoreState(store.getState());
+					const subscriber = store.on('change', handleStoreChange);
+
+					return () => subscriber.removeListener();
+				}
+			}, [store]);
+
+			return store ? (
+				<WrappedComponent
+					{...props}
+					{...mapStateToProps(storeState, props)}
+					{...mapDispatchToProps(store.dispatch, props)}
+				/>
+			) : (
+				<></>
```

Comment:
@p2kmgcl: I'm afraid we can't use this shorthand syntax until we update our version of the "jsdoc" package; I have a PR over with @petershin for that (https://github.com/petershin/liferay-portal/pull/587) but I need his help because I don't really know how all the Gradle plugins work.

Can you change this to `<React.Fragment></React.Fragment>`?

Reply:
Sure :+1: